src/core/namespace.c

```diff
@@ -89,6 +97,46 @@ static const TargetMount protect_kernel_tunables_table[] = {
         { "/sys/fs/cgroup",             READWRITE,      false, }, /* READONLY is set by ProtectControlGroups= option */
 };
 
+/* ProtectSystem=true table */
+static const TargetMount protect_system_yes_table[] = {
+        { "/usr",                       READONLY,       false,  },
+        { "/boot",                      READONLY,       true,   },
+        { "/efi",                       READONLY,       true,   },
+};
+
+/* ProtectSystem=full includes ProtectSystem=true */
+static const TargetMount protect_system_full_table[] = {
+        { "/usr",                       READONLY,       false,  },
+        { "/boot",                      READONLY,       true,   },
+        { "/efi",                       READONLY,       true,   },
+        { "/etc",                       READONLY,       false,  },
+};
+
+/*
+ * ProtectSystem=strict table. In this strict mode, we mount everything
+ * read-only, except for /proc, /dev, /sys which are the kernel API VFS,
+ * which are left writable, but PrivateDevices= + ProtectKernelTunables=
+ * protect those, and these options should be fully orthogonal.
+ * (And of course /home and friends are also left writable, as ProtectHome=
+ * shall manage those, orthogonally).
+ */
+static const TargetMount protect_system_strict_table[] = {
+        { "/",                          READONLY,       false,  },
+        { "/proc",                      READWRITE,      false,  },      /* ProtectKernelTunables= */
+        { "/sys",                       READWRITE,      false,  },      /* ProtectKernelTunables= */
+        { "/dev",                       READWRITE,      false,  },      /* PrivateDevices= */
+        { "/home",                      READWRITE,      true,   },      /* ProtectHome= */
+        { "/run/user",                  READWRITE,      true,   },      /* ProtectHome= */
+        { "/root",                      READWRITE,      true,   },      /* ProtectHome= */
+};
+
+static void append_mount(BindMount **p, const char *path, MountMode mode, bool ignore) {
```

Comment:
I don't think this function should modify *p. Maybe call it `configure_bind_mount` and let the caller update *p.